src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.cs

```diff
@@ -0,0 +1,2565 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Microsoft.UI.Private.Controls;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Uno.Extensions;
+using Uno.UI.Helpers.WinUI;
+using Windows.ApplicationModel;
+using Windows.Devices.Sensors;
+using Windows.Foundation;
+using Windows.Graphics.Display;
+using Windows.System;
+using Windows.UI.Composition;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Shapes;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class TeachingTip : ContentControl
+	{
+		private long m_automationNameChangedRevoker;
+		private long m_automationIdChangedRevoker;
+
+		public TeachingTip()
+		{
+			//__RP_Marker_ClassById(RuntimeProfiler.ProfId_TeachingTip);
+			DefaultStyleKey = typeof(TeachingTip);
+			Unloaded += ClosePopupOnUnloadEvent;
+			m_automationNameChangedRevoker = RegisterPropertyChangedCallback(AutomationProperties.NameProperty, OnAutomationNameChanged);
+			m_automationIdChangedRevoker = RegisterPropertyChangedCallback(AutomationProperties.AutomationIdProperty, OnAutomationIdChanged);
+			SetValue(TemplateSettingsProperty, new TeachingTipTemplateSettings());
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new TeachingTipAutomationPeer(this);
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			Dispatcher.AcceleratorKeyActivated -= OnF6AcceleratorKeyClicked;
+			EffectiveViewportChanged -= OnTargetLayoutUpdated;
+			if (m_tailOcclusionGrid != null)
+			{
+				m_tailOcclusionGrid.SizeChanged -= OnOcclusionContentChanged;
+			}
+			if (m_closeButton != null)
+			{
+				m_closeButton.Click -= OnCloseButtonClicked;
+			}
+			if (m_alternateCloseButton != null)
+			{
+				m_alternateCloseButton.Click -= OnCloseButtonClicked;
+			}
+			if (m_actionButton != null)
+			{
+				m_actionButton.Click -= OnActionButtonClicked;
+			}
+			var coreWindow = CoreWindow.GetForCurrentThread();
+			if (coreWindow != null)
+			{
+				coreWindow.SizeChanged -= WindowSizeChanged;
+			}
+
+			//IControlProtected controlProtected{ this };
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)